lib/sw_mapper.rb

```diff
@@ -149,37 +148,16 @@ def positive_int?(str)
     false
   end
 
+  # the @id attribute of resource/file elements that match the dor_content_type
   # value is used to tell SearchWorks UI app of specific display needs for objects
-  # a config file value for add_display_type can be used to prepend a string to
-  #  xxx_collection or xxx_object
-  # e.g., Hydrus objects are a special display case
-  # Based on a value of :add_display_type in a collection's config yml file
-  #
-  # information on DOR content types:
-  #   https://consul.stanford.edu/display/chimera/DOR+content+types%2C+resource+types+and+interpretive+metadata
-  #
-  # @return [String] 'file' or DOR content type
-  def display_type
-    if purlxml.dor_display_type.present?
-      purlxml.dor_display_type
-    else
-      case purlxml.dor_content_type
-      when 'book'
-        'book'
-      when 'image', 'manuscript', 'map'
-        'image'
-      else
-        'file'
-      end
-    end
-  end
-
-  # the @id attribute of resource/file elements that match the display_type
   # @return [Array<String>] filenames
   def file_ids
     return if purlxml.is_collection
-    return purlxml.image_ids if %w(image book).include?(display_type)
-    return purlxml.file_ids if display_type == 'file'
+    id = nil
+    filename = nil
+    filename = purlxml.image_ids.first if %w(book image manuscript map).include?(purlxml.dor_content_type)
+    id = filename.prepend("#{druid}%2F") unless filename.nil?
+    id
```

Comment:
this change looks good, but I think it could be tightened up:

```
filename = purlxml.image_ids.first if %w(book image manuscript map).include?(purlxml.dor_content_type)
return filename.prepend("#{druid}%2F") unless filename.nil?
```